packages/paste-core/components/menu/src/index.tsx

```diff
@@ -94,12 +107,51 @@ const Menu = React.forwardRef<HTMLDivElement, MenuProps>((props, ref) => {
 Menu.displayName = 'Menu';
 export {Menu};
 
-const MenuItem = React.forwardRef<HTMLDivElement, MenuItemProps>(({as = StyledMenuItem, ...props}, ref) => {
-  return <MenuPrimitiveItem {...props} as={as} ref={ref} />;
-});
+const MenuItem = React.forwardRef<HTMLDivElement, MenuItemProps>(
+  ({as = StyledMenuItem, variant: _variant, ...props}, ref) => {
+    const variant = _variant || React.useContext(MenuGroupContext);
+    return <MenuPrimitiveItem {...props} variant={variant} as={as} ref={ref} />;
+  }
+);
 MenuItem.displayName = 'MenuItem';
 export {MenuItem};
 
+const MenuGroup = React.forwardRef<HTMLDivElement, MenuGroupProps>(({label, icon, children, ...props}, ref) => {
+  return (
+    <MenuGroupContext.Provider value={MenuItemVariants.GROUP_ITEM}>
+      <Box {...safelySpreadBoxProps(props)} role="group" aria-label={label} textDecoration="none" ref={ref}>
```

Comment:
After doing some screen reader testing can you change the role to `presentation` for now? It causes the items to break if it's in a group at the moment. It was a hope more than anything that this would work.

Reply:
The inner Text has role presentation.. should I just remove role group here?

Reply:
For this one add the role of presentation explicitly _just in case_. I only tested it with presentation on this node and I can't be confident removing it won't introduce a regression. In theory it doesn't do anything, but _sometimes_ it's required to basically remove a node from the accessibility tree when it's in between 2 dependent roles.

So sometimes you might get

```
menu
  generic
    menuitem
    menuitem
```

and adding presentation to the div usually removes the generic

```
menu
  menuitem
  menuitem
```